### DISCUSSION TITLE
IDGEN Service - sequence number reset and custom format

### DISCUSSION THREAD

**Discussion by atulgupta2024** [egovernments/Digit-Core#403] [DIGIT Platform]
Q1. the ID formats that we use, has place holders for sequence number, for example
TenantID-INCIDENT_SQUENCE_NUMBER-YYYY
the understanding is that INCIDENT_SQUENCE_NUMBER is a rolling sequence number, defined in MDMS. What does this number reset? i.e. will it started from 1 again every year? Is that defined somewhere or is always 1 year reset window?

Q2. If I have a number like
LEGAL_ACT_CODE-CaseSequenceNumber-YYYY
where LEGAL_ACT_CODE is from another Master table, where legal acts (statutes) and sections are defined. Is this possible? How will the ID gen service know from which table to pull that code?

**Comment by kavi-egov** [comment]
ID Gen supports getting an ID by sending the name or the format. It replaces - boundary-code/city-code, financial year, current year, tenant-id and random text generator by default - https://core.digit.org/platform/core-services/id-generation-service
I would suggest that the utility to replace the legal act code should be written in the respective module and the format has to be updated before or after the ID-Gen call.

A Sequence can only be reset by query, creating a cron job as part of the migration that runs once a year should do it.   It can be implemented to look for sequences with certain name formats like SEQ_{RESETYEARLY}_CASE_CODE
